### PR TITLE
[develop] Update spack-stack on noaacloud

### LINKED ISSUE #881: SRW App doesn't build on PW NOAA Cloud Platforms
## Expected behavior
The SRW App should build on the NOAA Cloud platforms.

## Current behavior
Currently, the SRW App cannot be built on the NOAA Cloud platforms as the [modulefiles](https://github.com/ufs-community/ufs-srweather-app/blob/develop/modulefiles/build_noaacloud_intel.lua) is using an outdated spack-stack.

## Machines affected
PW NOAA Cloud platforms (i.e. Azure, AWS, and GCP)

## Steps To Reproduce
Follow the [document instructions](https://ufs-srweather-app.readthedocs.io/en/latest/BuildSRW.html) to build the app. The error will occur when building the app.

## Detailed Description of Fix (optional)
We were able to build the SRW App on the NOAA Cloud platforms using a newer spack-stack. See the noaa cloud module file here: /contrib/Edward.Snyder/rrfs-container-test/byhand/update/ufs-srweather-app/modulefiles.

## Possible Implementation (optional)
Ensure the changes listed above work for the other two cloud platforms and create a PR to push these changes to develop branch.

### PR DESCRIPTION
**Pull request by EdwardSnyder-NOAA**
## DESCRIPTION OF CHANGES: 
This PR updates the spack-stack used to build the SRW App to the latest spack-stack installed version. Changes also include using the new conda environment (workflow_tools), and updates the `NCORES_PER_NODE` variable to use the maximum compute cores for each Parallel Works Cloud Service Provider (PW CSPs).

### Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## TESTS CONDUCTED: 

- [ ] hera.intel
- [ ] orion.intel
- [ ] cheyenne.intel
- [ ] cheyenne.gnu
- [ ] gaea.intel
- [ ] jet.intel
- [ ] wcoss2.intel
- [X] NOAA Cloud (indicate which platform) Ran on all three PW CSPs
- [ ] Jenkins
- [x] fundamental test suite
All test pass except the MET VX test: `grid_RRFS_CONUS_25km_ics_NAM_lbcs_NAM_suite_GFS_v16.` That's because PR #826 was not tested on the NOAA Cloud platforms. Issue #905 was opened to track the progress of using the MET and METplus modules on NOAA Cloud.
- [ ] comprehensive tests (specify *which* if a subset was used)

## DOCUMENTATION:

No documentation updates are needed
## ISSUE: 

This addresses issue #881, which is cannot run the SRW App on NOAA Cloud

## CHECKLIST

- [x] My code follows the style guidelines in the Contributor's Guide
- [x] I have performed a self-review of my own code using the Code Reviewer's Guide
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [x] My changes do not require updates to the documentation (explain).
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

## LABELS (optional): 

A Code Manager needs to add the following labels to this PR: 
- [ ] Work In Progress
- [x] bug
- [ ] enhancement
- [ ] documentation
- [x] release
- [ ] high priority
- [ ] run_ci
- [ ] run_we2e_fundamental_tests
- [ ] run_we2e_comprehensive_tests
- [ ] Needs Cheyenne test 
- [ ] Needs Jet test 
- [ ] Needs Hera test 
- [ ] Needs Orion test 
- [ ] help wanted

## CONTRIBUTORS (optional): 

@natalie-perlin for installing workflow_tools on Azure and GCP.

### CODE REVIEW

**Review by natalie-perlin — APPROVED** [review]
MET and METplus modules in spack-stack have different environment variables. This may require SRW to be adapted to use MET and METplus modules

File: modulefiles/build_noaacloud_intel.lua [hunk 1]
```diff
@@ -5,13 +5,13 @@ the NOAA cloud using Intel-oneapi
...
 whatis([===[Loads libraries needed for building the UFS SRW App on NOAA cloud ]===])
 
-prepend_path("MODULEPATH", "/contrib/EPIC/spack-stack/envs/srw-develop-intel/install/modulefiles/Core")
+prepend_path("MODULEPATH", "/contrib/EPIC/spack-stack/spack-stack-1.4.1/envs/unified-dev/install/modulefiles/Core")
 prepend_path("MODULEPATH", "/apps/modules/modulefiles")
 prepend_path("PATH", "/contrib/EPIC/bin")
-load("intel/2021.3.0")
-load("impi/2021.3.0")
+--load("intel/2021.3.0")
+--load("impi/2021.3.0")
```
Comment (MichaelLueken) [inline modulefiles/build_noaacloud_intel.lua]: Will these commented out lines be used in the future?  If not, they should be removed to clean up the modulefile.

**Review by MichaelLueken — CHANGES_REQUESTED** [review]
@EdwardSnyder-NOAA - Overall these changes look good!

I noted that two lines, loading intel and impi, had been commented out in `build_noaacloud_intel.lua`.  If these will be used in the future, it is fine to keep them in and commented.  However, if they won't be used, please remove these two lines to help clean up the modulefile.

Additionally, there is a conflict in the `srw_common_spack.lua` modulefile.  Please merge the latest HEAD of `develop` into your `feature/update-spack-stack-noaacloud` branch and resolve the conflict.  Once complete, I can give my approval and move forward with merging this work.

File: modulefiles/build_noaacloud_intel.lua (hunk 1 above)
Comment (EdwardSnyder-NOAA) [reply modulefiles/build_noaacloud_intel.lua]: removed.

Comment (MichaelLueken) [reply modulefiles/build_noaacloud_intel.lua]: Thanks!

**Review by MichaelLueken — APPROVED** [review]
@EdwardSnyder-NOAA - Thanks for addressing the conflict and removing the comments from the modulefile!  Approving and merging this work now.

**Pull request merged**